### PR TITLE
Fix 116 - Remove Code Duplication

### PR DESCRIPTION
**Pull request by wxxedu**
Don't merge yet, not done. I created this just to have a look.

### CODE REVIEW

**Comment by wxxedu** [comment]
Ok, this part is done now. Can merge now.

**Comment by wxxedu** [comment]
Fixing #116

**Comment by wxxedu** [comment]
Ok there are some quite significant problems regarding the error message here. I will fix this.

**Comment by wxxedu** [comment]
The error messages have been fixed. I also discovered the problem with one of the previously closed issues. We will need to update documentation.

File: src/main/java/wingman/logic/flight/linklocation/FlightLocationLinkCommandFactory.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                locationMap
+        );
+
+        if (!hasLocation) {
+            throw ParseException.formatted(
+                    NO_LOCATION_MESSAGE,
+                    commandWord,
+                    getCommandFormatHint()
+            );
+        }
```
Comment (BoAi01) [inline src/main/java/wingman/logic/flight/linklocation/FlightLocationLinkCommandFactory.java:159]: Seems duplicated across different `LinkCommandFactory`s, but we can just leave them, given there are some class-specific differences here, e.g., `LOCATION_ARRIVAL_PREFIX` is unique to the Flight-Location link.

File: src/main/java/wingman/logic/pilot/linkflight/PilotFlightLinkCommandFactory.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        );
+
+        Map<FlightPilotType, Pilot> pilots = new HashMap<>();
+
+        boolean hasFoundPilot = addTarget(
+                PILOT_FLYING_PREFIX,
+                pilotFlyingIdOptional,
+                FlightPilotType.PILOT_FLYING,
+                pilots
+        ) | addTarget(
```
Comment (BoAi01) [inline src/main/java/wingman/logic/pilot/linkflight/PilotFlightLinkCommandFactory.java:149]: Nice use of `|`, as opposed to `||`

File: src/main/java/wingman/logic/pilot/linkflight/PilotFlightLinkCommandFactory.java [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+                pilots
+        );
+
+        if (!hasFoundPilot) {
+            throw ParseException.formatted(
+                    NO_PILOT_MESSAGE,
+                    commandWord,
+                    getCommandFormatHint()
+            );
+        }
```
Comment (BoAi01) [inline src/main/java/wingman/logic/pilot/linkflight/PilotFlightLinkCommandFactory.java:162]: Corresponding to the earlier comment.

File: src/main/java/wingman/logic/toplevel/link/LinkFactoryBase.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        Optional<I> flightOptional = manager.getItemOptional(flightId - 1);
+        if (flightOptional.isEmpty()) {
+            throw NumberParseException.formatted(
+                    "%s: with index %d is not found."
+                            + "\nCorrect Format: %s",
+                    getCommandWord(),
+                    flightId,
+                    getCommandFormatHint()
+            );
+        }
```
Comment (BoAi01) [inline src/main/java/wingman/logic/toplevel/link/LinkFactoryBase.java:112]: A little bit long, but this is not common in the entire code base. Then it should be good.

**Review by BoAi01 — APPROVED** [review]
Looks great to merge.

**Pull request merged**